### ISSUE TITLE
Refactor `ContentCardList` to use the new `KCardGrid` and the `AccessibleResourceCard` and `AccessibleFolderCard`

### ISSUE THREAD

**Issue by marcellamaki** [learningequality/kolibri#12732]
## Overview 

In the course of the work on #13008 and #12895, the refactoring of the content cards to use the new `AccessibleResourceCard` and `AccessibleFolderCard` was out of scope. However, to benefit from the accessibility improvements of the `KCard` in the new feature, we want to refactor the grid to use these new cards. 

### Requirements 

This issue focuses both on the actual refactor but also in-scope is _meticulous_ alignment to the figma specs if further refinements are needed. Now that we have a fully integrated side panel where we can use real data (content nodes) to stress test the content cards in a variety of screen resolutions and content types, we need to make sure the cards are pixel perfect and bug free, in addition to just swapping them out.

- [ ] REFACTOR: Use the `AccessibleFolderCard` and `AccessibleResourceCard` within the side panel for topic tree and search results navigation
- [ ] ADD: Card metadata is integrated and displayed according to the figma, using the composable created by Allan in #12686
- [ ] CONFIRM: Cards are properly responsive across the side panel display (on large screens) and compress properly on mobile screen resolutions
- [ ] CONFIRM: Cards are fully to spec. _Jessica should be added as a reviewer on the PR and give a 👍_

**Comment by marcellamaki** [comment]
~Managed in the course of other work~ It was not :)
